client/homebrew/brewRenderer/brewRenderer.jsx

```diff
@@ -35,8 +35,28 @@ const BrewPage = (props)=>{
 		index    : 0,
 		...props
 	};
+
+	const pageRef = useRef();
+
+	const rootMargin = '-50% 0px';
+
+	useEffect(()=>{
+		const observer = new IntersectionObserver(
+			([entry])=>{
+				const pageNo = parseInt(entry.target.getAttribute('id').slice(1));
+				updatePageArray(entry.isIntersecting, pageNo);
+			},
+			{ rootMargin }
+		);
+		observer.observe(pageRef.current);
+		return ()=>{
+			if(pageRef.current == null) return;
+			observer.unobserve(pageRef.current);
+		};
+	}, [pageRef.current, rootMargin]);
```

Comment:
Neither of these values trigger a redraw when changed, so they won't have any effect here in the `useEffect` dependency array. The effect will just fire once when the page component is mounted and then never again. Might as well just have an empty array here, no? `[]`